File: 0717. 1-bit and 2-bit Characters/Solution.py
from typing import List


class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        ones = 0
        i = len(bits) - 2
        while i >= 0 and bits[i]:
            ones += 1
            i -= 1
        return ones % 2 == 0

File: 0717. 1-bit and 2-bit Characters/Solution.java
class Solution {
    public boolean isOneBitCharacter(int[] bits) {
        int ones = 0;
        for (int i = bits.length - 2; i >= 0 && bits[i] == 1; i--)
            ones++;
        return ones % 2 == 0;
    }
}
